server/container_create_linux.go

```diff
@@ -330,17 +331,74 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 			specgen.AddMount(mnt)
 		}
 	}
-	mountLabel := sb.MountLabel()
-	processLabel := sb.ProcessLabel()
-	selinuxConfig := containerConfig.GetLinux().GetSecurityContext().GetSelinuxOptions()
-	if selinuxConfig != nil {
-		var err error
-		processLabel, mountLabel, err = getSELinuxLabels(selinuxConfig, privileged)
-		if err != nil {
+	imageSpec := containerConfig.GetImage()
+	if imageSpec == nil {
+		return nil, fmt.Errorf("CreateContainerRequest.ContainerConfig.Image is nil")
+	}
+
+	image := imageSpec.Image
+	if image == "" {
+		return nil, fmt.Errorf("CreateContainerRequest.ContainerConfig.Image.Image is empty")
+	}
+	images, err := s.StorageImageServer().ResolveNames(image)
+	if err != nil {
+		if err == storage.ErrCannotParseImageID {
+			images = append(images, image)
+		} else {
 			return nil, err
 		}
 	}
 
+	// Get imageName and imageRef that are later requested in container status
+	var (
+		imgResult    *storage.ImageResult
+		imgResultErr error
+	)
+	for _, img := range images {
+		imgResult, imgResultErr = s.StorageImageServer().ImageStatus(s.ImageContext(), img)
+		if imgResultErr == nil {
+			break
+		}
+	}
+	if imgResultErr != nil {
+		return nil, imgResultErr
+	}
+
+	var labelOptions []string
+	selinuxConfig := containerConfig.GetLinux().GetSecurityContext().GetSelinuxOptions()
+	if selinuxConfig == nil {
+		labelOptions = label.DupSecOpt(sb.ProcessLabel())
+	} else {
+		labelOptions = getLabelOptions(selinuxConfig, privileged)
```

Comment:
the privileged argument here is totally used and not checked in `getLabelOptions`, nor it's accounted in containers/storage. We need to fix and handle that by resetting processLabel to the empty string

Reply:
Removed

Reply:
but we need that to be used for CRI-O by erasing the processLabel